enhancements/builds/remove-jenkins-playload.md

```diff
@@ -0,0 +1,195 @@
+
+#### Telemetry / Operational Readiness
+
+The Jenkins imagestream will continue to be delivered to OpenShift via the Samples operator.
+If the Jenkins image fails to import, the existing ServiceMonitor for the Samples operator will alert that the imagestream failed to import after a reasonable period of time.
+These warning alerts will be sent to OpenShift's telemetry for CEE/SRE to debug.
+
+#### Documentation
+
+After Jenkins is removed from the payload, documentation related to Jenkins images will need to add mirroring instructions for disconnected clusters.
+Release notes will also need to inform admins that Jenkins images will need to be mirrored separately from the OpenShift payload.
+
+### Risks and Mitigations
+
+**Risk**: Jenkins imagestreams fail to import when the cluster is upgraded.
+
+*Mitigation*: Imagestreams don't report new references in status until the import is complete.
+Failures to import the new Jenkins imagestream will eventually fire an alert, unless the cluster admin skips the imagestream.
+
+**Risk**: Jenkins imagestreams are not updated in disconnected clusters.
+
+*Mitigation*: Documentation will need to provide new mirroring instructions for admins planning to upgrade.
+Release notes should also clearly indicate that Jenkins is no longer provided in the payload and needs to be mirrored separately.
+
+## Design Details
+
+### Open Questions [optional]
+
+None.
+
+### Test Plan
+
+This proposal is in the provisional state - test plan is not necessary at this time.
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+- What additional testing is necessary to support managed OpenShift service-based offerings?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+Not required - this continues the deprecation of a GA feature.
+
+#### Tech Preview -> GA
+
+Not required - this continues the deprecation of a GA feature.
+
+#### Removing a deprecated feature
+
+The Jenkins pipeline strategy has been officially deprecated since OCP 4.1.
+This proposal does not permanently disable the strategy - rather it changes how we deliver an on-top application.
+Documentation will be updated to reflect areas where customers are impacted (mainly for network restricted environments).
+
+### Upgrade / Downgrade Strategy
+
+On upgrade, the Jenkins imagestream will be updated to reflect the new non-payload image by the Samples operator.
```

Comment:
how does a cluster get a new jenkins image when they are not upgrading OCP itself?  Will the imagestream be configured for scheduled import so that it picks up changes automatically?    (that may be undesirable for some customers, so really it probably needs to be a choice)

(even when upgrading OCP, nothing will cause a re-import unless the imagestreamtag name that's being imported changes)

Reply:
@sbose78 does our Helm toolchain provide a means to upgrade a chart and update existing chart deployments, similar to what we have for OperatorHub? If so, this might be the best path forward (and would require us to deliver as a Helm chart instead of a template)

Reply:
i'm not sure how that would solve the problem of "new jenkins image pushed to the 4.9 tag, 4.9 clusters pick up the new image sha"?

Reply:
> does our Helm toolchain provide a means to upgrade a chart and update existing chart deployments, similar to what we have for OperatorHub?

Yes, it does. 

The chart authors would typically make a submission for the new version of the software, example, https://github.com/openshift-helm-charts/charts/tree/main/charts/partners/hashicorp/vault . Our UI/CLI experiences would pick this up and prompt users to upgrade if a newer version is available.

Reply:
@sbose78 will our Helm chart repo, like our OperatorHub repo, live in an index image that gets updated regularly?

Reply:
Verified - today's templates use imagechange triggers. On upgrade the imagestream tags should reimport, and clusters will pick up the new tags.